databricks/koalas/frame.py

```diff
@@ -950,6 +950,108 @@ def assign(self, **kwargs):
                            [name for name, _ in pairs if name not in self._metadata.column_fields]))
         return DataFrame(sdf, metadata)
 
+    def to_excel(self, excel_writer, sheet_name="Sheet1", na_rep="", float_format=None,
+                 columns=None, header=True, index=True, index_label=None, startrow=0,
+                 startcol=0, engine=None, merge_cells=True, encoding=None, inf_rep="inf",
+                 verbose=True, freeze_panes=None):
+        """
+        Write object to an Excel sheet.
+
+        To write a single object to an Excel .xlsx file it is only necessary to
+        specify a target file name. To write to multiple sheets it is necessary to
+        create an `ExcelWriter` object with a target file name, and specify a sheet
+        in the file to write to.
+
+        Multiple sheets may be written to by specifying unique `sheet_name`.
+        With all data written to the file it is necessary to save the changes.
+        Note that creating an `ExcelWriter` object with a file name that already
+        exists will result in the contents of the existing file being erased.
+
```

Comment:
This one would also need a note that says it will collect data into driver side which can end up with OOO

Reply:
I am not getting you here @HyukjinKwon, what do you mean by OOO?

Reply:
@rxin, @ueshin can you help me out here?

Reply:
@shril you should add this note.
```
.. note:: This method should only be used if the resulting DataFrame is expected
    to be small, as all the data is loaded into the driver's memory.
```

Reply:
Once again thanks @garawalid :)

Reply:
Peace!